### PR TITLE
Feat/map type toggle button

### PR DESCRIPTION
**Pull request by Ahmedhossamdev**

### CODE REVIEW

**Comment by Ahmedhossamdev** [comment]
image
image
image
image

File: src/lib/MapTypeButton/MapTypeButton.svelte [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        }
+    }
+</script>
+
+<button class="map-type-button" on:click={cycleMapType}>
+    {getMapTypeIcon(mapTypeId)}
+</button>
+
+<style>
+	.map-type-button {
```
Comment (tarunsinghofficial) [inline src/lib/MapTypeButton/MapTypeButton.svelte:37]: To maintain consistency, you can use the Tailwind CSS inline at the button.

**Review by tarunsinghofficial — CHANGES_REQUESTED** [review]
nicely done 👍🏻 .

File: src/lib/MapTypeButton/MapTypeButton.svelte (hunk 1 above)
Comment (tarunsinghofficial) [reply src/lib/MapTypeButton/MapTypeButton.svelte:37]: Okay, no problem. We will add the button CSS code into a global CSS file in the future.

**Comment by aaronbrethorst** [comment]
@Ahmedhossamdev please rebase against the latest changes in `main`

**Comment by Ahmedhossamdev** [comment]
@aaronbrethorst  Done

**Review by aaronbrethorst — APPROVED** [review]
looks good to me. I suspect we'll have to change some behavior of this down the road, but this is a solid and useful addition. thank you!

**Pull request merged**